File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
      /*  if(n <= 3)
            return n;
      
        
        int a = 3;
        int b = 2;
        for(int i = 0; i < n - 3; i++)
        {
        a = a + b;
        b = a - b;
        }
        return a;*/
        
        int prev = 0;
        int current = 1;
        int temp = 0;
        
        for(int i =0; i < n; i++)
        {
            temp = current;
            current = current + prev;
            prev = temp;
        }
        return current;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [0] * (n + 1)
        for i in range(1, n + 1):
            temp = 0
            for step in range(1,3):
                if(i - step > 0):
                    temp += dp[i - step]
                elif i - step == 0:
                    temp += 1
                dp[i] = temp
        return dp[n]        